### PR TITLE
Split into smaller files, add caching for most time-consuming operations

### PR DESCRIPTION
**Pull request by kapke**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -12,11 +12,12 @@
         "pretty": true,
         "strictNullChecks": true,
         "strict": true,
-        "lib": ["es5", "es2015", "es2016", "es2017"],
+        "lib": ["es5", "es2015", "es2016", "es2017", "dom"],
```
Comment (cparjaszewski) [inline tsconfig.json]: Are you sure `dom` needs to be the last lib? If not - how about sorting these and making "dom" a first element in the array? If now - how about small comment why "dom" need to be the last?

**Review by cparjaszewski — COMMENTED** [review]
Let you some comments

File: tsconfig.json (hunk 1 above)
Comment (kapke) [reply tsconfig.json]: 1. JSON doesn't allow comments
2. To be honest - it's not needed anymore and it's removed already

File: src/spec/DeceiverMirror.spec.ts [hunk 2]
```diff
@@ -137,116 +135,27 @@ describe('DeceiverMirror', () => {
...
-        class Foo {}
-        class Bar {}
 
-        class B {
-            constructor (private bar: Bar) {}
-
-            public abc () {}
+    describe('caching', () => {
+        class TestClass {
+            public get foo () {throw new Error('Foo'); }
```
Comment (kamilpodlasek) [inline src/spec/DeceiverMirror.spec.ts]: missing space before `throw`

**Review by kamilpodlasek — APPROVED** [review]

File: src/spec/DeceiverMirror.spec.ts (hunk 2 above)
Comment (kapke) [reply src/spec/DeceiverMirror.spec.ts]: Prettier is going to fix this

**Pull request merged**